app/src/main/java/com/jnj/vaccinetracker/visitsoverview/screens/VisitsListFragment.kt

```diff
@@ -153,7 +157,8 @@ class VisitsListFragment(private val visitsKey: String) : BaseFragment(),
 
     private fun exportToExcel(visits: List<VisitDataDTO>) {
         val fileName = "${buildFileName()}.xls"
-        val mimeType = "application/vnd.ms-excel"
+        val mimeType = "application/vnd.openxmlformats-officedocument.spreadsheetml.sheet"
+        val file = File(requireContext().getExternalFilesDir(null), fileName)  // Use external storage
```

Comment:
does this make sense to pass null to getExternalFilesDir? as far as I remember these files are saved in Downloads directory? Do you know how does it work here?